finagle-http/src/main/scala/com/twitter/finagle/http/exp/ServerDispatcher.scala

```diff
@@ -41,11 +41,15 @@ private[finagle] abstract class GenSerialServerDispatcher[Req, Rep, In, Out](tra
    * [[com.twitter.finagle.http.codec.HttpServerDispatcher]].
    */
   protected def dispatch(req: Out): Future[Rep]
-  protected def handle(rep: Rep): Future[Unit]
+  protected def handle(rep: Rep, keepAlive: Boolean): Future[Unit]
+
+  protected def isReqKeepAlive(req: Out): Boolean = true
+  protected def isRepKeepAlive(rep: Rep): Boolean = true
 
   private[this] def loop(): Future[Unit] = {
     state.set(Idle)
     trans.read().flatMap { case Multi(req, eos) =>
+      val reqKeepAlive = isReqKeepAlive(req)
```

Comment:
This  intentional.  Requests are mutable, and so I want to capture the keepalive state before the request is dispatched to a service that may, for instance, remove the Connection header to send it somewhere else.